src/main/java/org/mitre/synthea/engine/Generator.java

```diff
@@ -334,52 +382,242 @@ private synchronized void writeToConsole(Person person, int index, long time, bo
+      }
+
+      //Birthplace
+      for (int i = 0; i <= birthplaces.size() - 1; i++) {
+        if (randSpew.get("POBP").equals(birthplaces.get(i).get("Code"))) {
+          person.attributes.put(Person.BIRTHPLACE, birthplaces.get(i).get("Pob"));
+        }
+      }
+
+      String schoolEnrollment = randSpew.get("SCH");
+
+      if (schoolEnrollment.equals("NA")) {
+        person.attributes.put(Person.SCHOOL_ENROLLMENT, "N/A (less than 3 years old)");
+      } else if (schoolEnrollment.equals("1")) {
+        person.attributes.put(Person.SCHOOL_ENROLLMENT, "no");
+      } else if (schoolEnrollment.equals("2")) {
+        person.attributes.put(Person.SCHOOL_ENROLLMENT, "public_school_or_public_college");
+      } else if (schoolEnrollment.equals("3")) {
+        person.attributes.put(Person.SCHOOL_ENROLLMENT, "private_school_or_college_or_home_school");
+      }
+
+      //Grade level
+      for (int i = 0; i <= gradeLevels.size() - 1; i++) {
+        if (randSpew.get("SCHG").equals(gradeLevels.get(i).get("Code"))) {
+          person.attributes.put(Person.GRADE_LEVEL, gradeLevels.get(i).get("grade"));
+        }
+      }
+
+      //Relationship
+      for (int i = 0; i <= relationships.size() - 1; i++) {
+        if (randSpew.get("RELP").equals(relationships.get(i).get("Code"))) {
+          person.attributes.put(Person.RELATIONSHIP, relationships.get(i).get("Relationship"));
+        }
+      }
+
+      // Socioeconomic variables of education, income, and education are set.
+      String education = city.pickEducation(person.random);
+      person.attributes.put(Person.EDUCATION, education);
+      double educationLevel = city.educationLevel(education, person);
+      person.attributes.put(Person.EDUCATION_LEVEL, educationLevel);
+
+      long targetAge = Long.valueOf(randSpew.get("AGEP")).longValue();
+
+      int income = Integer.parseInt(randSpew.get("PINCP"));
+      person.attributes.put(Person.INCOME, income);
+
+      double incomeLevel = city
+          .incomeLevel(Integer.parseInt(randSpew.get("HINCP")));
+      person.attributes.put(Person.INCOME_LEVEL, incomeLevel);
+
+      double occupation = person.rand();
+      person.attributes.put(Person.OCCUPATION_LEVEL, occupation);
+
+      double sesScore = city.socioeconomicScore(incomeLevel, educationLevel, occupation);
+      person.attributes.put(Person.SOCIOECONOMIC_SCORE, sesScore);
+      person.attributes.put(Person.SOCIOECONOMIC_CATEGORY, city.socioeconomicCategory(sesScore));
+
+      //Occupation
+      for (int i = 0; i <= occupations.size() - 1; i++) {
+        if (randSpew.get("OCCP").equals(occupations.get(i).get("Code"))) {
+          person.attributes.put(Person.OCCUPATION, occupations.get(i).get("Occupation"));
+        }
+      }
+
+      String employmentStatus = randSpew.get("ESR");
+
+      if (employmentStatus.equals("NA")) {
+        person.attributes.put(Person.EMPLOYMENT_STATUS, "na_under_16");
+      } else if (employmentStatus.equals("")) {
```

Comment:
Just realized these strings are all empty and need to be filled in with the right codes.

The last thing I'll mention is that these big if/else checks can be kind of ugly. There are a couple different ways to do these but fundamentally it's a judgment call so do whatever you prefer.

One way is to use a `switch` statement which you can do on Strings since Java 7 I think:

```java
      String employmentStatusCode = spewPerson.get(randSpew).get("ESR");
      String employmentStatusDesc;
      switch (employmentStatusCode) {
      case "NA":
        employmentStatusDesc = "na_under_16";
        break;
      case "1":
        employmentStatusDesc = "civilian_employed_at_work";
        break;
      case "2":
        employmentStatusDesc = "civilian_employed_with_job_but_not_at_work";
        break;
        // ... etc
        default:
          throw new IllegalArgumentException("Unexpected SPEW employment code: " + employmentStatusCode);
      }
      person.attributes.put(Person.EMPLOYMENT_STATUS, employmentStatusDesc);
```

Another way is to define the lookup table somewhere else and just call it, for an example see LifecycleModule.DIABETES_DRUG_HBA1C_IMPACTS . Then you just can do something along the lines of
```java
    String employmentStatusCode = spewPerson.get(randSpew).get("ESR");
    String employmentStatusDesc = SPEW_EMPLOYMENT_STATUS_CODES.get(employmentStatusCode);
    person.attributes.put(Person.EMPLOYMENT_STATUS, employmentStatusDesc);
```